tests/test_integration.py

```diff
@@ -0,0 +1,161 @@
+    backward_speed = init_speed.backward_estimation(max_fit=estimates_speed, tolerance=1e-1, max_events=2)
+    fig, ax = plt.subplots(1)#captures plots
+    hmp.visu.plot_topo_timecourse(epoch_data, backward_speed, info, init_speed, ax=ax)
+    #LOOCV
+    loocv_model_speed = hmp.loocv.loocv(init_speed, hmp_speed_data, backward_speed, print_warning=False, verbose=False)
+    #Same but testing multiproc.
+    init_speed = hmp.models.hmp(hmp_speed_data, epoch_data, sfreq=epoch_data.sfreq, cpus=2)
+    backward_speed = init_speed.backward_estimation(max_fit=estimates_speed, tolerance=1e-1, max_events=2)
+    #LOOCV
+    loocv_model_speed = hmp.loocv.loocv(init_speed, hmp_speed_data, backward_speed, print_warning=False, verbose=False)
+    #hmp.visu.plot_loocv(loocv_model_speed, pvals=True, test='sign', indiv=True, mean=True)
+
+    # the magnitudes map indicates the events. We have two conditions, speed and accuracy, hence two rows. For speed the third event is missing, indicated by the -1.
+    mags_map = np.array([[0, -1],
+                        [0, 0]])
+
+    # the parameters maps indicates the stages. Here, we indicate that stage 3 is missing for the speed condition. Obviously, this has to be congruent with magnitudes map.
+    pars_map = np.array([[0, -1, 0],
+                        [0, 0, 0]])
+
+    # finally, we have to define the conditions we want to analyze:
+    conds = {'cue': ['SP', 'AC']} #dictionary with conditions to analyze as well as the levels.
+    # we take the starting parameters from the accuracy model (you could also take the average)
+    mags4 = backward_speed.sel(n_events=2).magnitudes.dropna('event').data
+    pars4 = backward_speed.sel(n_events=2).parameters.dropna('stage').data
+
+    fig, ax = plt.subplots(1)#captures plots
+    init = hmp.models.hmp(hmp_data, epoch_data, sfreq=epoch_data.sfreq, cpus=1)
+    #fit the model - note that we use the full data again
+    
+    model_stage_removed = init.fit_single_conds(magnitudes=mags4, parameters=pars4, pars_map=pars_map, mags_map=mags_map, conds=conds,  cpus=1, tolerance=1e-1, verbose=False)
+    hmp.visu.plot_topo_timecourse(epoch_data, model_stage_removed, info, init, magnify=1, sensors=False, time_step=1000/init.sfreq,xlabel='Time (ms)', event_lines=True, colorbar=True, title="Remove one event",ax=ax) 
+    correct_loocv_model = hmp.loocv.loocv_backward(init, hmp_data, max_events=2)
+    #hmp.visu.plot_loocv(correct_loocv_model[0], pvals=True, test='sign', indiv=True, mean=True)
+
+    hmp.utils.save_fit(selected, 'selected.nc')
+    hmp.utils.save_eventprobs(selected.eventprobs, 'selected_eventprobs.nc')
+    estimates = hmp.utils.load_fit('selected.nc')
+    loocv_combined = hmp.loocv.loocv(init, hmp_data, model_stage_removed, print_warning=False)
+    #hmp.visu.plot_latencies(model_stage_removed, init, errs='se',kind='bar')
+
+for filename in os.listdir("tests"):
```

Comment:
The files won't be created in the tests directory though, at least in my experience. I would hard code the files that are generated instead, so that there is no chance that something gets deleted that was not meant to be deleted.

Reply:
Weirdly I can't delete them because I don't know where they are created during the tests...